Shared/Supporting Files/Views/SampleLink.swift

```diff
@@ -131,11 +131,18 @@ private extension String {
     /// - Parameter substring: The substring to bold.
     /// - Returns: The attributed string with the bolded substring.
     func boldingFirstOccurrence(of substring: String) -> AttributedString {
-        if let range = localizedStandardRange(of: substring.trimmingCharacters(in: .whitespacesAndNewlines)),
-           let boldedString = try? AttributedString(markdown: replacingCharacters(in: range, with: "**\(self[range])**")) {
-            return boldedString
-        } else {
-            return AttributedString(self)
+        var attributedString = AttributedString(self)
+        
+        let trimmedSubstring = substring.trimmingCharacters(in: .whitespacesAndNewlines)
+        if let range = localizedStandardRange(of: trimmedSubstring) {
+            let substring = self[range]
+            
+            if let boldedSubstring = try? AttributedString(markdown: "**\(substring)**"),
+               let attributedRange = attributedString.range(of: substring) {
```

Comment:
```suggestion
            if let boldedSubstring = try? AttributedString(markdown: "**\(substring)**"),
               let attributedRange = attributedString.range(of: self[range]) {
```
Since the local variable `substring` is only used here, move it here can avoid the potential confusion with the method argument.

Reply:
It's actually needed in `AttributedString(markdown: "**\(substring)**")` as well, but I replaced both uses of it so I could apply your following suggestion.

Reply:
Oops, apparently I misinterpreted which variable is which as well 😓